File: neoforge/src/main/java/com/eliasnvx/krylix/forge/client/PlayerStatsClient.java
package com.eliasnvx.krylix.forge.client;

import com.eliasnvx.krylix.forge.network.NetworkPackets.PlayerStatEntry;

import java.util.ArrayList;
import java.util.List;

public class PlayerStatsClient {
    private static List<PlayerStatEntry> entries = new ArrayList<>();

    public static void updateStats(List<PlayerStatEntry> newEntries) {
        entries = newEntries;
    }

    public static List<PlayerStatEntry> sortedByKills() {
        List<PlayerStatEntry> sorted = new ArrayList<>(entries);
        sorted.sort((a, b) -> Integer.compare(b.kills(), a.kills()));
        return sorted;
    }

    public static List<PlayerStatEntry> sortedByMobKills() {
        List<PlayerStatEntry> sorted = new ArrayList<>(entries);
        sorted.sort((a, b) -> Integer.compare(b.mobKills(), a.mobKills()));
        return sorted;
    }
}

File: forge/src/main/kotlin/com/eliasnvx/krylix/forge/client/PlayerStatsClient.kt
package com.eliasnvx.krylix.forge.client

import com.eliasnvx.krylix.forge.network.NetworkPackets.PlayerStatsSyncPacket.PlayerStatEntry

/**
 * Клиентский кэш PvP-рейтинга, приходящего с сервера через PlayerStatsSyncPacket
 * (per-world PlayerKillStatsData, см. серверный аналог).
 */
object PlayerStatsClient {
    private var entries: List<PlayerStatEntry> = emptyList()

    fun updateStats(newEntries: List<PlayerStatEntry>) {
        entries = newEntries
    }

    fun sortedByKills(): List<PlayerStatEntry> = entries.sortedByDescending { it.kills }

    fun sortedByMobKills(): List<PlayerStatEntry> = entries.sortedByDescending { it.mobKills }
}
